lnwallet/wallet.go

```diff
@@ -69,6 +69,13 @@ type InitFundingReserveMsg struct {
 	// workflow.
 	NodeAddr net.Addr
 
+	// LocalSpendAmt should be set to the amount we intend to spend when
+	// opening the channel. This can be used for instance to use all our
+	// remaining funds to open the channel, since it will take fees into
+	// account. Can currently only be used when LocalFundingAmt and
+	// RemoteFundingAmt are set to 0.
+	LocalSpendAmt btcutil.Amount
```

Comment:
WTB enum structs. An alternative to make it clear only one of these can be specified would be something similar to what's shown in https://play.golang.org/p/jfgsCBxb9q.

Reply:
Another approach would be to make this a `SubtractFees bool` and keep `LocalFundingAmt` the same. Then we can just thread the boolean down to coin selection.

Reply:
Yeah, a enum/union holding the amount and the type of amount would be ideal here. Defining a new type seemed a bit overkill.

Using the bool would work, would you prefer it over the current approach? (code wise it would be more or less the same).

Reply:
Yeah I think the bool approach is cleaner.